wdl2cwl/tests/wdl_files/tasks/tasks_reports.wdl

```diff
@@ -0,0 +1,535 @@
+    maxRetries: 2
+  }
+}
+
+task assembly_bases {
+    meta {
+      description: "Count bases in a fasta file."
+    }
+
+    input {
+      File   fasta
+      String docker ="ubuntu"
+    }
+
+    command {
+        set -e
+        grep -v '^>' "~{fasta}" | tr -d '\n' | wc -c | tee assembly_length
+        grep -v '^>' "~{fasta}" | tr -d '\nNn' | wc -c | tee assembly_length_unambiguous
+    }
+
+    output {
+        Int assembly_length             = read_int("assembly_length")
+        Int assembly_length_unambiguous = read_int("assembly_length_unambiguous")
+    }
+
+    runtime {
+        docker: "${docker}"
+        memory: "1 GB"
+        cpu: 1
+        disks: "local-disk 50 HDD"
+        dx_instance_type: "mem1_ssd1_v2_x2"
+        maxRetries: 2
+    }
+}
+
+task fastqc {
+  input {
+    File   reads_bam
+
+    String docker = "quay.io/broadinstitute/viral-core:2.1.33"
+  }
+
+  String   reads_basename=basename(reads_bam, ".bam")
+
+  command {
+    set -ex -o pipefail
+    reports.py --version | tee VERSION
+    reports.py fastqc ${reads_bam} ${reads_basename}_fastqc.html --out_zip ${reads_basename}_fastqc.zip
+  }
+
+  output {
+    File   fastqc_html      = "${reads_basename}_fastqc.html"
+    File   fastqc_zip       = "${reads_basename}_fastqc.zip"
+    String viralngs_version = read_string("VERSION")
+  }
+
+  runtime {
+    memory: "2 GB"
+    cpu: 1
+    docker: "${docker}"
+    disks: "local-disk 375 LOCAL"
+    dx_instance_type: "mem1_ssd1_v2_x2"
+    maxRetries: 2
+  }
+}
+
+task align_and_count {
+  input {
+    File   reads_bam
+    File   ref_db
+    Int    topNHits = 3
+
+    Int?   machine_mem_gb
+    String docker = "quay.io/broadinstitute/viral-core:2.1.33"
+  }
+
+  String  reads_basename=basename(reads_bam, ".bam")
+  String  ref_basename=basename(ref_db, ".fasta")
+
+  command {
+    set -ex -o pipefail
+
+    read_utils.py --version | tee VERSION
+
+    ln -s "${reads_bam}" "${reads_basename}.bam"
+    read_utils.py minimap2_idxstats \
+      "${reads_basename}.bam" \
+      "${ref_db}" \
+      --outStats "${reads_basename}.count.${ref_basename}.txt.unsorted" \
+      --loglevel=DEBUG
+
+    sort -b -r -n -k3 "${reads_basename}.count.${ref_basename}.txt.unsorted" > "${reads_basename}.count.${ref_basename}.txt"
+    head -n ${topNHits} "${reads_basename}.count.${ref_basename}.txt" > "${reads_basename}.count.${ref_basename}.top_${topNHits}_hits.txt"
+    head -1 "${reads_basename}.count.${ref_basename}.txt" | cut -f 1 > "${reads_basename}.count.${ref_basename}.top.txt"
+  }
+
+  output {
+    File   report           = "${reads_basename}.count.${ref_basename}.txt"
+    File   report_top_hits  = "${reads_basename}.count.${ref_basename}.top_${topNHits}_hits.txt"
+    String top_hit_id       = read_string("${reads_basename}.count.${ref_basename}.top.txt")
+    String viralngs_version = read_string("VERSION")
+  }
+
+  runtime {
+    memory: "15 GB"
+    cpu: 4
+    docker: docker
```

Comment:
This doesn't make sense, `docker` is not the name of a valid container

Shouldn't it be what was in the original file?
```suggestion
    docker: "~{docker}"
```

Reply:
The original file was a string with placeholders that I haven't implemented. I will do that now.